File: packetCapture/src/main/java/com/ninpou/packetcapture/core/util/common/Shells.java
package com.ninpou.packetcapture.core.util.common;

import java.io.IOException;
import java.util.*;

public class Shells {
    public static String getDns() {
        Scanner scanner = null;
        try {
            Process process = Runtime.getRuntime().exec("getprop net.dns1");
            scanner = new Scanner(process.getInputStream());
            return scanner.nextLine();
        } catch (IOException e) {
            return null;
        } finally {
            IOUtils.close(scanner);
        }
    }
}

File: packetCapture/src/main/java/com/ninpou/packetcapture/core/util/common/Shells.kt
package com.ninpou.packetcapture.core.util.common

import java.io.IOException
import java.util.*

object Shells {
    val dns: String?
        get() {
            var scanner: Scanner? = null
            return try {
                val process = Runtime.getRuntime().exec("getprop net.dns1")
                scanner = Scanner(process.inputStream)
                scanner.nextLine()
            } catch (e: IOException) {
                null
            } finally {
                IOUtils.close(scanner)
            }
        }
}